python/lsst/fgcmcal/fgcmBuildStarsBase.py

```diff
@@ -0,0 +1,810 @@
+    """
+    Base task to build stars for FGCM global calibration
+    """
+    def __init__(self, butler=None, **kwargs):
+        """
+        Instantiate an `FgcmBuildStarsBaseTask`.
+
+        Parameters
+        ----------
+        butler : `lsst.daf.persistence.Butler`
+        """
+
+        pipeBase.CmdLineTask.__init__(self, **kwargs)
+        self.makeSubtask("sourceSelector")
+        # Only log warning and fatal errors from the sourceSelector
+        self.sourceSelector.log.setLevel(self.sourceSelector.log.WARN)
+
+    # no saving of metadata for now
+    def _getMetadataName(self):
+        return None
+
+    @pipeBase.timeMethod
+    def runDataRef(self, butler, dataRefs):
+        """
+        Cross-match and make star list for FGCM Input
+
+        Parameters
+        ----------
+        butler:  `lsst.daf.persistence.Butler`
+        dataRefs: `list` of `lsst.daf.persistence.ButlerDataRef`
+           Source data references for the input visits.
+           If this is an empty list, all visits with src catalogs in
+           the repository are used.
+           Only one individual dataRef from a visit need be specified
+           and the code will find the other source catalogs from
+           each visit.
+
+        Raises
+        ------
+        RuntimeErrror: Raised if `config.doReferenceMatches` is set and
+           an fgcmLookUpTable is not available, or if computeFluxApertureRadius()
+           fails if the calibFlux is not a CircularAperture flux.
+        """
+        self.log.info("Running with %d src dataRefs" % (len(dataRefs)))
+
+        if self.config.doReferenceMatches:
+            # Ensure that we have a LUT
+            if not butler.datasetExists('fgcmLookUpTable'):
+                raise RuntimeError("Must have fgcmLookUpTable if using config.doReferenceMatches")
+        # Compute aperture radius if necessary.  This is useful to do now before
+        # any heavy lifting has happened (fail early).
+        calibFluxApertureRadius = None
+        if self.config.doSubtractLocalBackground:
+            sourceSchema = butler.get('src_schema').schema
+            try:
+                calibFluxApertureRadius = computeApertureRadius(sourceSchema,
+                                                                self.config.instFluxField)
+            except (RuntimeError, LookupError):
+                raise RuntimeError("Could not determine aperture radius from %s. "
+                                   "Cannot use doSubtractLocalBackground." %
+                                   (self.config.instFluxField))
+
+        groupedDataRefs = self.findAndGroupDataRefs(butler, dataRefs)
+
+        camera = butler.get('camera')
+
+        # Make the visit catalog if necessary
+        # First check if the visit catalog is in the current path
+        visitCatDataRef = butler.dataRef('fgcmVisitCatalog')
+        filename = visitCatDataRef.get('fgcmVisitCatalog_filename')[0]
+        if os.path.exists(filename):
```

Comment:
I've added a fuller comment that this trickery is to make sure that it only looks for the visit catalog in the current directory.  Otherwise it looks in parents as well, and so you can't have the resume functionality _and_ the ability to do reruns.